pkg/telemetry/telemetry.go

```diff
@@ -18,10 +17,13 @@ const (
 )
 
 type Telemetry struct {
+	sink    CloseableMetricSink
 	Metrics *metrics.Metrics
 	labels  []metrics.Label
 }
 
+var _ Closeable = (*Telemetry)(nil)
```

Comment:
In order to support consistent emission of metrics we now keep a reference to the metric sink (which must be closeable) and expose a Close() method on Telemetry which then closes the underlying sink.